course_discovery/apps/course_metadata/data_loaders/api.py

```diff
@@ -899,3 +899,98 @@ def _update_program_banner_image(self, body, program):
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=self.max_workers) as executor:
+            if self.is_threadsafe:
+                for page in pagerange:
+                    time.sleep(30)
+                    executor.submit(self._load_data, page)
+            else:
+                for future in [executor.submit(self._make_request, page) for page in pagerange]:
+                    response = future.result()
+                    self._process_response(response)
+
+        logger.info('Retrieved %d course from %s.', count, self.partner.marketing_site_api_url)
+
+    def _make_request(self, page):
+        """
+        Send request to WordPress.
+        """
+        logger.info('Requesting WordPress course run page %d...', page)
+        params = {'page': page, 'page_size': self.PAGE_SIZE}
+        auth = requests.auth.HTTPBasicAuth(settings.WORDPRESS_USERNAME, settings.WORDPRESS_APP_PASSWORD)
+        response = requests.get(self.api_url, auth=auth, params=params)
+        response.raise_for_status()
+        return response.json()
+
+    def _load_data(self, page):
+        """
+        Make a request for the given page and process the response.
+        """
+        response = self._make_request(page)
+        self._process_response(response)
+
+    def _process_response(self, response):
+        """
+        Process the response from the WordPress.
+        """
+        results = response['results']
+        logger.info('Retrieved %d WordPress course runs...', len(results))
+
+        for body in results:
+
+            course_run_key = body['course_id']
+            try:
+                body = self.clean_strings(body)
+                course_run = CourseRun.objects.get(key__iexact=course_run_key)
+                course_run.short_description_override = body['excerpt']
+                course_run.full_description_override = body['description']
+                course_run.featured = body['featured']
+                course_run.card_image_url = body['featured_image_url']
+                course_run.slug = body['slug']
+                category = body['category']
+
+                course_run.tags.clear()
+                for tag in body['tags']:
+                    course_run.tags.add(tag)
+
+                subject = Subject.objects.filter(
+                    partner=self.partner
+                ).filter(slug=category['slug']).first()
+                if not subject:
+                    subject = Subject.objects.create(
+                        name=category['title'],
+                        description=category['description'],
+                        slug=category['slug'],
+                        partner=self.partner
+                    )
+                course_run.course.subjects.clear()
+                course_run.course.subjects.add(subject)
+                course_run.save()
+            except Exception:  # pylint: disable=broad-except
```

Comment:
can there be any specific exception that could occur?

Reply:
`DoesNotExist` only if the somehow the course exist in WordPress but not in Discovery

Reply:
In that case, I would suggest using a specific exception then instead of catching the generic exception.

Reply:
@iam-mhaseeb  @rehan99000  specific exception added.